### DISCUSSION TITLE
How to span a single bar across multiple displays

### DISCUSSION THREAD

**Discussion by FancyFurret** [FelixKratz/SketchyBar#410] [Q&A]
Hi! I recently got an ultrawide monitor, and this monitor is able to act as two separate displays, side by side. In order to get full resolution out of this monitor, I need to use this mode. I have disabled "Displays have separate spaces", so that I can position windows over this "split". The next issue is that the native macos bar is only on one half of my monitor (again, since it is acting as two separate displays to macos, a left half and right half). 

So, I was hoping that by using a custom bar, I would be able to span it across two displays, so that I have one long bar going across my entire monitor. I can't seem to find a way of doing this in the configuration though. I am able to restrict the bar to just the main display, and now i just need to override it's width or something to get it to be wide enough.

Is something like this possible? If so, how?

**Comment by FelixKratz** [comment]
Sure, this is something that can be achieved.

The idea is to embrace the fact that you have two displays and two separate bars for the displays. Each item in sketchybar can be assigned to specific displays or spaces via the `associated_display` option (see https://felixkratz.github.io/SketchyBar/config/items#geometry-properties).

With this you can add any item you would like to the "left half" of your bar and the "right half" of the bar (display 1 and 2) respectively, such that you can design the bar as if it was one continous bar across the two displays.

Example:
```bash
sketchybar --bar height=30 color=0xff000000

sketchybar --add item item1 left \
           --set item1 icon="An item on the left half" \
                       associated_display=1

sketchybar --add item item2 right \
           --set item2 icon="An item on the right half" \
                       associated_display=2
```

**Reply by FancyFurret** [reply]
Ah got it, that makes sense. I guess the only issue there is it would be nice if I could put the clock in the center. I could put it on the left/right of the right/left bar, but it would be slightly off center.
